source/extensions/transport_sockets/tls/context_config_impl.cc

```diff
@@ -352,6 +370,14 @@ ClientContextConfigImpl::ClientContextConfigImpl(
        config.common_tls_context().tls_certificate_sds_secret_configs().size()) > 1) {
     throw EnvoyException("Multiple TLS certificates are not supported for client contexts");
   }
+  auto optional_validation_context = getValidationContext(config.common_tls_context());
+  if (optional_validation_context.has_value()) {
+    if (!optional_validation_context.value().has_trusted_ca() &&
+        !config.allow_dangerous_tls_without_certificate_validation()) {
+      // FIXME warn for today.
```

Comment:
I guess there should be a runtime flag to control this is a warning message or an exception raising?

I know this is for backward-compatible. But it is strange you have field called 'allow_something', but when it is false, it is just a warning.